app/models/exam_template.rb

```diff
@@ -221,13 +221,21 @@ def save_pages(partial_exams)
             division.start <= page_num && page_num <= division.end
           end
         end
+        extra_pages.sort_by! { |page_num, _| page_num }
         extra_pdf = CombinePDF.new
-        extra_pdf << extra_pages.collect { |_, page| page }
+        cover_pdf = CombinePDF.new
+        cover_pdf << extra_pages[0][1]
```

Comment:
You need to check whether the first page here is actually 1 or not. (Could be that the first page was missing.)